js/embed_media_back_button.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file
+ * Adds the back button for embeded media select modal in CKEditor.
+ * Everytime the user goes to different page within the media embed select modal
+ * it will save the filter values to retrive the search result when user
+ * comes back to the first page.
+ */
+
+ (function ($, Drupal) {
+  'use strict';
+
+  Drupal.behaviors.tide_media_browser_iframe = {
+    attach: function (context, settings) {
+      if (!window.CKEDITOR) {
+        return;
+      }
+      let iframePath, itemName, licenseType, mediaType, published, site
+      const iframeName = 'entity_browser_iframe_tide_media_browser_iframe'
+      const homePagePath = '/entity-browser/iframe/tide_media_browser_iframe'
+      const backButton = $('#entity-embed-dialog-tide-media-browser-home')
+      $(document).ready(function () {
+        // Remove the name field value on modal first load if any.
+        localStorage.removeItem("tideMediaBrowsernameFilterVal");
+        $("iframe").on("load", function() {
+          if ($("iframe") && $("iframe").length > 0) {
+            for (let i = 0; i < $("iframe").length; i++) {
+              if ($("iframe")[i].name && $("iframe")[i].name === iframeName) {
+                iframePath = $("iframe")[1].contentWindow.location.pathname
+                // Shows the back button when iframe is not in the first page.
+                if (iframePath && iframePath !== homePagePath) {
+                  backButton.css("display", "block");
+                }
+                // Hides the back button for the first page.
+                if (iframePath && iframePath === homePagePath) {
+                  backButton.css("display", "none");
+                }
+                // Get the filter values on iframe load.
+                itemName =  $(this).contents().find('#edit-name').val()
+                licenseType = $(this).contents().find('#edit-field-license-type-target-id-1 :selected').text()
```

Comment:
I'm not seeing this saved in local storage. Think your selector might not be quite right for all but the name field. e.g. this license type has this HTML:

```
<select data-drupal-selector="edit-field-license-type-target-id-1" id="edit-field-license-type-target-id-1--1_-ORPXVgy4" name="field_license_type_target_id_1" class="form-select">
```

Your id is `edit-field-license-type-target-id-1--1_-ORPXVgy4` which looks a bit random. You might need to find the name attribute seems more consistent.

Reply:
1. The filter values except for name field doesn't need to be saved in local storage and coming back to the first page the selection remains the same. Only the namefield value needs to be stored.
2. so looks like JQuery can target partial id, got the idea from here - https://stackoverflow.com/questions/37300558/how-to-select-elements-using-jquery-by-partial-id-and-a-class-at-the-same-time
and this scenario it works as well and targets the partial id

Reply:
> Hey @anthony-malkoun the drop down saves fine when I tried in the testing PR now. See the attached steps I followed - [live media embed demo.webm](https://user-images.githubusercontent.com/20810541/178217544-bb28165e-e3ab-4300-80c4-36687a7df501.webm) 

Hmm, that's not quite what I'm seeing. If I have some filters applied, view an item and hit previous, I go back to the listing (at page 1) and the filters are selected but I have to hit apply again to get them active. Is this as expected?

https://user-images.githubusercontent.com/4596219/178373973-3b7e3c77-dab3-475b-ab2e-865e87a6c4b9.mp4